modules/data-loading/partials/kafka/kafka-data-source-details.adoc

```diff
@@ -13,62 +13,239 @@ To configure the data source object, the minimum requirement is the address of t
+
+If both source and target clusters are sharing the same SSL settings, user can set generic settings for both source/target clusters and all the rols(admin/producer/consumer). 
+
+For example, user can set `ssl.keystore.location="/path/to/key/store"` instead of:
+
+* `source.cluster.ssl.keystore.location="/path/to/key/store"`
+* `admin.ssl.keystore.location="/path/to/key/store"`
+* `source.cluster.admin.ssl.keystore.location="/path/to/key/store"`.
+
+If source and target clusters have different SSL settings, it is possible to set cluster wide SSL configs. 
+
+For example, user can set: `target.cluster.ssl.truststore.password="/password/for/trust/store"` instead of:
+
+* `target.cluster.producer.ssl.trust.password="/password/for/trust/store"`.
+
+Note: SSL is now well supported by TigerGraph, we recommend users to set up regular SSL rather than SASL + PlainText/SSL.
+
+==== Supported Configuration Examples
+===== PLAINTEXT
+[source,json,linenum]
+----
+{
+  "type": "mirrormaker",
+  "source.cluster.bootstrap.servers": "<KAFKAURL>:<PORT>"
+}
+----
+
+===== SSL
+Need to configure:
+
+* `<prefix>.security.protocol` 
+* `<prefix>.ssl.<field>`
+
+[source,json,linenum]
+----
+{
+  "type": "mirrormaker",
+  "source.cluster.bootstrap.servers": "<KAFKAURL>:<PORT>",
+
+  "consumer.security.protocol": "SSL",
+  "consumer.ssl.endpoint.identification.algorithm": "none",
+  "consumer.ssl.keystore.location": "/path/to/client.keystore.jks",
+  "consumer.ssl.keystore.password": "******",
+  "consumer.ssl.key.password": "******",
+  "consumer.ssl.truststore.location": "/path/to/client.truststore.jks",
+  "consumer.ssl.truststore.password": "******",
+
+  "source.admin.security.protocol": "SSL",
+  "source.admin.ssl.endpoint.identification.algorithm": "none",
+  "source.admin.ssl.keystore.location": "/path/to/client.keystore.jks",
+  "source.admin.ssl.keystore.password": "******",
+  "source.admin.ssl.key.password": "******",
+  "source.admin.ssl.truststore.location": "/path/to/client.truststore.jks",
+  "source.admin.ssl.truststore.password": "******",
+
+  "producer.security.protocol": "SSL",
+  "producer.ssl.endpoint.identification.algorithm": "none",
+  "producer.ssl.keystore.location": "/path/to/client.keystore.jks",
+  "producer.ssl.keystore.password": "******",
+  "producer.ssl.key.password": "******",
+  "producer.ssl.truststore.location": "/path/to/client.truststore.jks",
+  "producer.ssl.truststore.password": "******"
+}
+----
+
+===== SASL_PLAINTEXT
+Need to configure:
+
+* `<prefix>.security.protocol`
+* `<prefix>.sasl.<field>`
+
+[source,json,linenum]
+----
+{
+  "type": "mirrormaker",
+  "source.cluster.bootstrap.servers": "<KAFKAURL>:<PORT>",
+
+  "consumer.security.protocol": "SASL_PLAINTEXT",
+  "consumer.sasl.mechanism": "<GSSAPI/PLAIN/etc>",
+
+  "source.admin.security.protocol": "SASL_PLAINTEXT",
+  "source.admin.sasl.mechanism": "<GSSAPI/PLAIN/etc>",
```

Comment:
@pingxieTG  is the `<prefix>.sasl.jaas.config` only needed for third party  security protocol? Or would it also be necessary for the default?

Reply:
`sasl.jaas.config` is required for all mechanisms including the built-ins.
For example: https://docs.confluent.io/platform/current/security/authentication/sasl/gssapi/overview.html#clients
Here the GSSAPI has the following config:
```
sasl.jaas.config=com.sun.security.auth.module.Krb5LoginModule required \
    useKeyTab=true \
    storeKey=true \
    keyTab="/etc/security/keytabs/kafka_client.keytab" \
    principal="kafkaclient1@EXAMPLE.COM";
```